Chess/Assets/Scripts/ArrayChess/PieceMover.cs

```diff
@@ -54,83 +49,54 @@ public class PieceMover : MonoBehaviour
     private Vector3[] chessSquaresDistance;
 
     private float targetDistance = 1f;
-    private float loadtimer = 1f;
 
     void Start()
     {
         rend = GetComponent<Renderer>();
         pieceStartColor = rend.material.color;
-
+        chessSquares = GameObject.FindGameObjectsWithTag("square"); // Populates the chessboard array of all square positions 
     }
 
-
-
-
-
     void Update()
     {
-        if (!isInitialSquareRendererLoaded)
-        {
-            getSquareRenderers();
-        }
-
-        // add a if to load only once per turn.
+        //cacheSquareData(); TODO: Will be info on previous square, for now will be default colour
 
-
-        if (!isHover && isBoardArrayLoaded && isSquareRendLoaded && isColourUpdated) // Reset colour of piece and pieceSquare.
-        {
-
-            resetColors(false);
-            isColourUpdated = false;
-
-        }
-
-        if (!isBoardArrayLoaded) // Loads chessBoard.
+        if (Input.GetMouseButton(0)) // All permutations of mouse clicks go here
         {
-
-            findAllSquares();
-            isBoardArrayLoaded = true;
-        }
-
-        if (Input.GetMouseButton(0) && isSquareRendLoaded) // Clicking piece locks colour of movesquare until another piece is clicked.
-        {
-            //cacheSquareData();
             ray = Camera.main.ScreenPointToRay(Input.mousePosition);
             Physics.Raycast(ray, out hit);
-            rayHitStringParent = hit.transform.parent.name;
-            getSquareRenderers();
-            isCached = true;
-            
+            mouseClickPosition = hit.transform.parent.name;
 
+            updateNearestSquareCoords();
+            updateMovementSquare();
 
-            if (isShowingMovement && rayHitStringParent == moveSquare.transform.parent.name) // Move piece
-            {
+            var isChessPieceClicked = mouseClickPosition == piece.transform.name;
```

Comment:
Added these extra variables for readability. Notice that these variables are scoped within this update function and cant be used outside of it.